-1

Tested at: f2b71a278387932bee6e8219e7bc61ffd202537f


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #27585 @bsunanda: Phase2-hgx207 Have uniform wafer type from HGCalDDDConst and reduce usage of magic numbers
 - #27588 @ianna: DD4hep:Simulation - DDDWorld Defines Regions and Sets G4 Production Cuts
 - #27520 @mtosi: TRK POG Phase2 : limit highPurity tracks up to |eta| < 4.0
 - #27505 @jpata: PFAlgo thread-safe logging
 - #27574 @cvuosalo: Enhancements to support the magnetic field geometry builder

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fe51d/1629/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fe51d/1629/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fe51d/1629/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDDWorld.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4SensitiveConverter.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4ProductionCuts.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/SensitiveDetectorCatalog.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4ProductionCuts.cc: In member function 'void DDG4ProductionCuts::initialize()':
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4ProductionCuts.cc:85:29: error: request for member 'toString' in '((DDG4ProductionCuts*)this)->DDG4ProductionCuts::map_', which is of pointer type 'const G4LogicalVolumeToDDLogicalPartMap* {aka const DDMapper<G4LogicalVolume*, DDLogicalPart>*}' (maybe you meant to use '->' ?)
     unsigned int num = map_.toString(keywordRegion_, vv.second, regionName);
                             ^~~~~~~~
gmake: *** [tmp/slc7_amd64_gcc700/src/SimG4Core/Geometry/src/SimG4CoreGeometry/DDG4ProductionCuts.cc.o] Error 1
>> Compiling bigobj /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4SolidConverter.cc 
>> Compiling bigobj /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4Builder.cc 
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDGeometryReturnType.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDDWorld.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4SensitiveConverter.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4ProductionCuts.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/SensitiveDetectorCatalog.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-24-2300/src/SimG4Core/Geometry/src/DDG4ProductionCuts.cc:85:28: error: member reference type 'const G4LogicalVolumeToDDLogicalPartMap *' (aka 'const DDMapper<G4LogicalVolume *, DDLogicalPart> *') is a pointer; did you mean to use '->'?
    unsigned int num = map_.toString(keywordRegion_, vv.second, regionName);
                       ~~~~^
                           ->
1 error generated.
gmake: *** [tmp/slc7_amd64_gcc700/src/SimG4Core/Geometry/src/SimG4CoreGeometry/DDG4ProductionCuts.cc.o] Error 1
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - #27585 @bsunanda: Phase2-hgx207 Have uniform wafer type from HGCalDDDConst and reduce usage of magic numbers
 - #27588 @ianna: DD4hep:Simulation - DDDWorld Defines Regions and Sets G4 Production Cuts
 - #27520 @mtosi: TRK POG Phase2 : limit highPurity tracks up to |eta| < 4.0
 - #27505 @jpata: PFAlgo thread-safe logging
 - #27574 @cvuosalo: Enhancements to support the magnetic field geometry builder

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fe51d/1629/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-7fe51d/1629/git-merge-result

--- a/SimG4Core/Geometry/src/DDG4ProductionCuts.cc
+++ b/SimG4Core/Geometry/src/DDG4ProductionCuts.cc
@@ -82,7 +82,7 @@ void DDG4ProductionCuts::initialize() {
   G4Region* region = nullptr;
   G4RegionStore* store = G4RegionStore::GetInstance();
   for (auto const& vv : vec_) {
-    unsigned int num = map_.toString(keywordRegion_, vv.second, regionName);
+    unsigned int num = map_->toString(keywordRegion_, vv.second, regionName);
     edm::LogVerbatim("Geometry") << "  num  " << num << " regionName: " << regionName << ", the store of size "
                                  << store->size();
     if (num != 1) {

